Set mysql2 adapter prepared_statements default to true

stefannibrasil opened the pull request:
### Summary

This is the final step into fixing [prepared statements is not enabled by default with mysql2 adapter](https://github.com/rails/rails/issues/42975).

This PR changes the default prepared_statements value to true for the MySQL2 adapter.

It's the same behavior that others adapters have.

Co-authored-by Thiago Araujo @thdaraujo

---

## Question
Would adding a note about this [configuration on the guides](https://guides.rubyonrails.org/configuring.html#configuring-a-postgresql-database) be helpful? If so, we could add a follow up PR.

sbs-cats committed:
Set mysql2 prepared_statements default to true

byroot commented:
Are you aware of https://github.com/rails/rails/issues/43005? I think this issue would need to be fixed before the default is changed.

stefannibrasil commented:
> Are you aware of #43005? I think this issue would need to be fixed before the default is changed.

good call, thanks for bringing this up! @byroot 
I guess we'll have to wait until #43005 is fixed until we merge this PR.

This PR is also waiting on a [previous PR](https://github.com/rails/rails/pull/43141) as well, so should be fine. It was approved, we're just waiting for it to be merged.

Thanks!

yahonda commented:
Reopened this pull request because I think it is worth discussing this pull request as Rails stops autoclosing pull requests via https://github.com/rails/rails/commit/acf48169943011834c4c885b956e29050548fa98

stefannibrasil commented:
thanks @yahonda [I posted this](https://discuss.rubyonrails.org/t/mysql2-adapter-prepared-statements-default-set-to-true-next-steps/79906) in the Rails forum hoping to get some ideas on how to move forward with this one, if anyone is interested.

sbs-cats committed:
Set MySQL2 prepared_statements default to true

Fixes #42975

When prepared_statements was not set for the MySQL2
adapter, the default value was false.

This behavior was deprecated in #43141.

The default prepared_statement for MySQL2 is
now set to true.

Co-authored-by: Thiago Araujo <thd.araujo@gmail.com>

sbs-cats committed:
Reverting change to config example test

sbs-cats committed:
Add entry to CHANGELOG

sbs-cats committed:
Remove deprecation test

Co-authored-by: Thiago Araujo <thd.araujo@gmail.com>

sbs-cats committed:
Fix changelog linter error

Co-authored-by: Thiago Araujo <thd.araujo@gmail.com>
